pkg/backends/influxdb/backend.go

```diff
@@ -231,91 +232,79 @@ func (b *Backend) writeWorkouts(workouts []*healthautoexport.Workout, targetName
+			},
+		}, healthautoexport.Field{
+			Key: "elevation_descent",
+			Value: &healthautoexport.QtyWithUnit{
+				Qty:   workout.Elevation.Descent,
+				Units: workout.Elevation.Units,
+			},
+		})
 	}
-
 	// Add other WorkoutFields
-	for name, field := range workout.Fields {
-		workoutFields[name] = field
-	}
-
+	workoutFields = append(workoutFields, workout.Fields...)
 	// Convert to InfluxDB field format
-	fields := MakeInfluxFieldsFromWorkoutFields(workoutFields)
-
+	for _, field := range workoutFields {
+		fieldName := GetUnitizedMeasurementName(field.Key, field.Value)
+		point.AddField(fieldName, float64(field.Value.Qty))
+	}
 	// Skip if there are no fields to write
-	if len(fields) == 0 {
+	if len(point.FieldList()) == 0 {
 		return nil, nil
 	}
-
-	point := write.NewPoint("workout", tags, fields, workout.Start.Time)
+	point.SetTime(workout.Start.Time)
 	return point, nil
 }
 
 func (b *Backend) createWorkoutPoints(
-	name string, tags map[string]string, data []*healthautoexport.DatapointWithUnit,
+	name string, tags []lp.Tag, data []*healthautoexport.DatapointWithUnit,
 ) []*write.Point {
 	points := make([]*write.Point, 0, len(data))
 	for _, datum := range data {
-		measurement := GetUnitizedMeasurementName(name, datum)
-		fields := map[string]interface{}{
-			"qty": float64(datum.Qty),
-		}
-		point := write.NewPoint(measurement, tags, fields, datum.Date.Time)
+		point := write.NewPointWithMeasurement(GetUnitizedMeasurementName(name, datum))
+		addTagsToPoint(point, tags)
+		point.AddField("qty", float64(datum.Qty))
+		point.SetTime(datum.Date.Time)
 		points = append(points, point)
 	}
 	return points
 }
 
 func (b *Backend) createRoutePoints(
-	name string, tags map[string]string, data []*healthautoexport.RouteDatapoint,
+	name string, tags []lp.Tag, data []*healthautoexport.RouteDatapoint,
 ) []*write.Point {
 	points := make([]*write.Point, 0, len(data))
 	for _, datum := range data {
-		measurement := name
-		fields := map[string]interface{}{
-			"lat":      datum.Lat,
-			"lon":      datum.Lon,
-			"altitude": datum.Altitude,
-		}
-		point := write.NewPoint(measurement, tags, fields, datum.Timestamp.Time)
+		point := write.NewPointWithMeasurement(name)
+		addTagsToPoint(point, tags)
+		point.AddField("lat", datum.Lat)
+		point.AddField("lon", datum.Lon)
+		point.AddField("altitude", datum.Altitude)
+		point.SetTime(datum.Timestamp.Time)
 		points = append(points, point)
 	}
 	return points
 }
 
-// MakeTags returns a map of tags that can be safely modified.
-// Accepts a targetName, which if not empty, will be added to the map.
-func (b *Backend) MakeTags(additional map[string]string) map[string]string {
-	tags := make(map[string]string, len(b.staticTags))
-	for k, v := range b.staticTags {
-		if v != "" {
-			tags[k] = v
-		}
-	}
-	for k, v := range additional {
-		if v != "" {
-			tags[k] = v
-		}
+func addTagsToPoint(point *write.Point, tags []lp.Tag) {
+	for _, tag := range tags {
+		point.AddTag(tag.Key, tag.Value)
```

Comment:
Good catch. I don't even use that feature (yet?), so that might have broken for me. I've added that back, and also added an integration test for this. 

Please note i'm not 100% confident in new lines 93/94 of backend_test.go - note the trailing comma after measurement name which doesn't appear to match the [line protocol docs](https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/). But the point we are sending the influx client has nil tags confirmed in the debugger, so I'm assuming that's a client implementation detail?

Reply:
It seems that [`PointToLineProtocolBuffer`](https://github.com/grafana/influxdb-client-go/blob/v1.5.0/api/write/ext.go#L18) is indeed implemented this way, I guess we don't have to worry about it.